providers/vic/proxy/isolation_proxy.go

```diff
@@ -440,45 +443,79 @@ func (v *VicIsolationProxy) SetState(op trace.Operation, handle, name, state str
 	defer trace.End(trace.Begin(id, op))
 
 	if v.client == nil {
-		return "", vicerrors.NillPortlayerClientError("IsolationProxy")
+		return vicerrors.NillPortlayerClientError("IsolationProxy")
 	}
 
-	results, err := v.client.Containers.GetContainerInfo(containers.NewGetContainerInfoParamsWithContext(op).WithID(id))
+	pForce := force
+	params := containers.NewContainerRemoveParamsWithContext(op).
+		WithID(id).
+		WithForce(&pForce).
+		WithTimeout(120 * time.Second)
+
+	removeOK, err := v.client.Containers.ContainerRemove(params)
+	op.Debugf("ContainerRemove returned %# +v", removeOK)
+	return err
+}
+
+func (v *VicIsolationProxy) State(op trace.Operation, id, name string) (string, error) {
+	defer trace.End(trace.Begin(id, op))
+
+	payload, err := v.getInfo(op, id, name)
 	if err != nil {
-		switch err := err.(type) {
-		case *containers.GetContainerInfoNotFound:
-			return "", vicerrors.NotFoundError(name)
-		case *containers.GetContainerInfoInternalServerError:
-			return "", vicerrors.InternalServerError(err.Payload.Message)
-		default:
-			return "", vicerrors.InternalServerError(fmt.Sprintf("Unknown error from the interaction port layer: %s", err))
-		}
+		return "", err
 	}
 
-	state := results.Payload.ContainerConfig.State
+	state := payload.ContainerConfig.State
 	return state, nil
 }
 
-func (v *VicIsolationProxy) Remove(op trace.Operation, id string, force bool) error {
+func (v *VicIsolationProxy) NetworkConfigs(op trace.Operation, id, name string) ([]*models.EndpointConfig, error) {
+	defer trace.End(trace.Begin(id, op))
+
+	payload, err := v.getInfo(op, id, name)
+	if err != nil {
+		return nil, err
+	}
+
+	return payload.Endpoints, nil
+}
+
+func (v *VicIsolationProxy) PodConfig(op trace.Operation, id, name string) (*models.ContainerConfig, error) {
+	defer trace.End(trace.Begin(id, op))
+
+	payload, err := v.getInfo(op, id, name)
+	if err != nil {
+		return nil, err
+	}
+
+	return payload.ContainerConfig, nil
+}
+
+// Private methods
+func (v *VicIsolationProxy) getInfo(op trace.Operation, id, name string) (*models.ContainerInfo, error) {
 	defer trace.End(trace.Begin(id, op))
 
 	if v.client == nil {
-		return vicerrors.NillPortlayerClientError("IsolationProxy")
+		return nil, vicerrors.NillPortlayerClientError("IsolationProxy")
 	}
 
-	pForce := force
-	params := containers.NewContainerRemoveParamsWithContext(op).
-		WithID(id).
-		WithForce(&pForce).
-		WithTimeout(120 * time.Second)
+	results, err := v.client.Containers.GetContainerInfo(containers.NewGetContainerInfoParamsWithContext(op).WithID(id))
+	if err != nil {
+		switch err := err.(type) {
+		case *containers.GetContainerInfoNotFound:
+			return nil, vicerrors.NotFoundError(name)
+		case *containers.GetContainerInfoInternalServerError:
+			return nil, vicerrors.InternalServerError(err.Payload.Message)
+		default:
+			return nil, vicerrors.InternalServerError(fmt.Sprintf("Unknown error from the interaction port layer: %s", err))
```

Comment:
this might be a copy paste issue.  We should remove 'interaction' because it's not actually the interaction port layer.  I think portlayer is enough.